src/pnpq/devices/odl_ozoptics_650ml.py

```diff
@@ -151,24 +153,23 @@ def serial_send(self, serial_cmd):
         self.conn.flushOutput()  # flush output buffer, aborting current output and discard all that is in buffer
         self.conn.write(serial_cmd.encode())
 
-    def serial_read(self, retries=10):
+    def serial_read(self):
         # The Python serial "in_waiting" property is the count of bytes available
         # for reading at the serial port.  If the value is greater than zero
         # then we know we have content available.
         device_output = ""
-        got_OK = False
-        while self.conn.in_waiting > 0 or (got_OK is False and retries > 0):
-            device_output += (self.conn.read(self.conn.in_waiting)).decode("iso-8859-1")
-            # command output is complete.
-            if device_output.find("Done") >= 0:
-                got_OK = True
-            time.sleep(0.05)
-            retries -= 1
+        device_output += (self.conn.read_until(expected=bytearray(b"Done"))).decode(
+            "iso-8859-1"
+        )
+        if "Done" in device_output:
+            self.logger.debug("Device read successful: %s", device_output)
+        else:
+            raise RuntimeError("Reading from the device failed (Timeout)!")
```

Comment:
This if/else statement confuses two orthogonal actions:

1. Debug logging all captured device output
2. Validating that the read completed successfully.

It's better to just run the debug log statement all of the time.

And we might want to see partial outputs even if debug logging is turned off, so adding it to the RuntimeError might be a good idea.

```suggestion
        self.logger.debug("Device read output: %s", device_output)
        if "Done" not in device_output:
            raise RuntimeError(f"Device read did not complete before timeout. Read output was: {device_output}")
```

Reply:
Done!